### PR TITLE
Version control module

### LINKED ISSUE #40: Version control module
# Version Control

Basics of what the module should cover: 

- What is version control  
- Why to use it 
- When in the research cycle to use it 
- Difference between local and remote version control 
- Methods/tools for version control (Git/Github/GitLab/SVN/Mercurial) 
- Link to resources that explain the basics of how commonly used version features (such as branches, pull, push, commit, merge) work 

## Learning Objectives (learning outcomes)

- Explain what version control of research software is  

- Illustrate the importance of using version control  

- Identify when in the research cycle to employ version control 

- Compare the use of local and remote version control  

- List the various tools or systems for version control  

- Identify relevant resources for researchers about features commonly used in version control  

## Exercises

Exercise 1: Case study: A researcher comes to you with a problem of how to keep track of different version of code. The question to answer: what advice would you give the researcher to keep track of changes of their scripts/code and when best to do this (objective 1, 2, 3 and 4); would you advise them to work with a local or remote version control system 

Exercise 2: In breakout groups of 3-4 people, identify 3 commonly used version control systems or tools that you have heard of (or used yourself) (for example: Git). Together, search for a resource (such a website, or handbook) for each tool that helps explain how to use its key features (meets objectives 5,6)  

##Resources  

[Version Control with Git: Automated Version Control (swcarpentry.github.io)](https://swcarpentry.github.io/git-novice/01-basics.html) 

https://the-turing-way.netlify.app/reproducible-research/vcs 

https://coderefinery.github.io/git-intro/

### PR DESCRIPTION
**Pull request by JaroCamphuijsen**
This PR:

-  incorporates the changes from #43 
- fix #40 
-  merge main branch and solve conflicts
- make changes to adhere to updated NEBULA
- add learning objectives
- Add a slide deck as introduction

### CODE REVIEW

**Comment by JaroCamphuijsen** [comment]
This PR also updates some thumbnails, as I didn't want to open a new PR just for these tiny changes.

File: modules/version-control/version_control_exercise1.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+title: Case Study
+type: exercise
+order: 2
+---
+
+## Exercise: Case Study
+
+**Case study**: A researcher comes to you with a problem of how to keep track of different versions of code. 
+
+What advice would you give the researcher to keep track of changes of their scripts/code and when best to do this (objective 1, 2, 3 and 4).
```
Comment (raar1) [inline modules/version-control/version_control_exercise1.md:11]: Is this a reference to specific objectives discussed elsewhere or is this more "structure your answer as a list of objectives"?

File: modules/version-control/slides-version-control.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
+---
```
Comment (raar1) [inline modules/version-control/slides-version-control.md:1]: I like this glossary page

File: modules/version-control/version_control_exercise2.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+title: Tool search
+type: exercise
+order: 3
+---
+
+## Exercise: tool search
+
+In breakout groups of 3-4 people, identify 3 commonly used version control systems or tools that you have heard of (or used yourself) (for example: Git).
```
Comment (raar1) [inline modules/version-control/version_control_exercise2.md:9]: Is this meaning mercurial, svn, fossil etc? Are most students of this course likely to have even heard of one other than git?

**Review by raar1 — APPROVED** [review]
The "Documents are..." slide doesn't work consistently with loading the fragment. I would suggest getting rid of the fragment tags for now as there might be some clash with the styling.

The key points slide is missing a final bullet point. Or is that intentional?

Everything else looks really nice!

File: modules/version-control/version_control_exercise1.md (hunk 1 above)
Comment (JaroCamphuijsen) [reply modules/version-control/version_control_exercise1.md:11]: I think it's the first. I will take it out as it is not interesting to the learner.

File: modules/version-control/version_control_exercise2.md (hunk 3 above)
Comment (JaroCamphuijsen) [reply modules/version-control/version_control_exercise2.md:9]: I'll open an issue to revise both exercises.

**Comment by JaroCamphuijsen** [comment]
#117 is supposed to pick up where this PR ends

**Pull request merged**